src/mesh.cpp

```diff
@@ -1,80 +1,116 @@
+  vtkSmartPointer<vtkGenericDataObjectReader> reader =
+      vtkSmartPointer<vtkGenericDataObjectReader>::New();
+  reader->SetFileName(filename.c_str());
+  reader->SetReadAllScalars(true);
+  reader->SetReadAllVectors(true);
+  reader->ReadAllFieldsOn();
+  reader->Update();
+
+  // Get Points
+  vtkPoints *Points = reader->GetUnstructuredGridOutput()->GetPoints();
+  vtkIdType  NumPoints =
+      reader->GetUnstructuredGridOutput()->GetNumberOfPoints();
+  double vertexPos[3];
+  for (vtkIdType point = 0; point < NumPoints; point++) {
+    Points->GetPoint(point, vertexPos);
+    std::array<double, 3> vertexPosArr{vertexPos[0], vertexPos[1],
+                                       vertexPos[2]};
+    mesh.positions.push_back(vertexPosArr);
   }
-  std::ifstream connFile{filename};
-  std::string   line;
-  while (std::getline(connFile, line)) {
-    std::vector<std::string> parts;
-    boost::split(parts, line, [](char c) { return c == ' '; });
-    std::vector<size_t> indices(parts.size());
-    std::transform(parts.begin(), parts.end(), indices.begin(), [](const std::string &s) -> size_t { return std::stol(s); });
-
-    if (indices.size() == 3) {
-      std::array<size_t, 3> elem{indices[0], indices[1], indices[2]};
-      mesh.triangles.push_back(elem);
-    } else if (indices.size() == 2) {
-      std::array<size_t, 2> elem{indices[0], indices[1]};
-      mesh.edges.push_back(elem);
-    } else {
-      throw std::runtime_error{std::string{"Invalid entry in connectivitiy file \""}.append(line).append("\"")};
+  // Get Point Data
+  vtkPointData *PD = reader->GetUnstructuredGridOutput()->GetPointData();
+  // Check it has data array
+  if (PD->HasArray(dataname.c_str()) == 1) {
+    // Get Data and Add to Mesh
+    vtkDataArray *ArrayData = PD->GetArray(dataname.c_str());
+    int           NumComp   = ArrayData->GetNumberOfComponents();
+    switch (NumComp) {
+    case 1: // Scalar Data
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        const double scalar = ArrayData->GetTuple1(tupleIdx);
+        mesh.data.push_back(scalar);
+      }
+      break;
+    case 2: // Vector Data with 2 component
+      double *vector2ref;
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        vector2ref = ArrayData->GetTuple2(tupleIdx);
+        mesh.data.push_back(vector2ref[0]);
+        mesh.data.push_back(vector2ref[1]);
+      }
+      break;
+    case 3: // Vector Data with 3 component
+      double *vector3ref;
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        vector3ref = ArrayData->GetTuple3(tupleIdx);
+        mesh.data.push_back(vector3ref[0]);
+        mesh.data.push_back(vector3ref[1]);
+        mesh.data.push_back(vector3ref[2]);
+      }
+      break;
     }
+  } else { // There is no data in mesh file fill with zeros.
+    std::clog << "There is no data found. Dummy data will be used!.\n";
```

Comment:
```suggestion
    std::clog << "No data found was found in mesh $meshname. The data \"\ $dataname" will be filled with zero data values.\n";
```
Do you associate the dummy data values with a datanme? Maybe add both, the data name and the meshname in the log message.

Reply:
Data name is added. Actually, I am not sure using std::clog is the correct way to do this. ? Should I create some other logging methods for the verbose log like summary() method?

Reply:
I would just use `cout`